tests/PHPStan/Rules/Methods/CallMethodsRuleTest.php

```diff
@@ -34,9 +47,69 @@ protected function getRule(): Rule
 	{
 		$reflectionProvider = $this->createReflectionProvider();
 		$ruleLevelHelper = new RuleLevelHelper($reflectionProvider, $this->checkNullables, $this->checkThisOnly, $this->checkUnionTypes, $this->checkExplicitMixed, $this->checkImplicitMixed, true, false);
+
 		return new CallMethodsRule(
 			new MethodCallCheck($reflectionProvider, $ruleLevelHelper, true, true),
-			new FunctionCallParametersCheck($ruleLevelHelper, new NullsafeCheck(), new PhpVersion($this->phpVersion), new UnresolvableTypeHelper(), new PropertyReflectionFinder(), true, true, true, true, true),
+			new FunctionCallParametersCheck(
+				$ruleLevelHelper,
+				new NullsafeCheck(),
+				new PhpVersion($this->phpVersion),
+				new UnresolvableTypeHelper(),
+				new PropertyReflectionFinder(),
+				new ParameterClosureTypeHelper(
+					new class implements ParameterClosureTypeExtensionProvider
+					{
+
+						public function getFunctionParameterClosureTypeExtensions(): array
+						{
+							return [];
+						}
+
+						public function getMethodParameterClosureTypeExtensions(): array
+						{
+							return [
+								new class implements MethodParameterClosureTypeExtension
+								{
+
+									public function isMethodSupported(MethodReflection $methodReflection, ParameterReflection $parameter): bool
+									{
+										return $methodReflection->getName() === 'foo' && $methodReflection->getDeclaringClass()->getName() === 'MethodParameterClosureTypeExtension\\Foo';
```

Comment:
When I run `make phpstan` locally it complains about `$methodReflection->getDeclaringClass()->getName() === 'MethodParameterClosureTypeExtension\\Foo'` being always false. Perhaps PHPStan is overly strict in this case? The code is fine IMO. It's especially annoying that it doesn't happen in Github Actions, so I can't even ignore it by adding it to the baseline.

Reply:
Try running `composer dump` first and then `make phpstan` again. What happens?

Reply:
I haven't tried this exactly, but I rebased it on top of 1.12.x, ran `composer install` and now it doesn't complain anymore. So you were probably right.